src/tribler/core/components/restapi/rest/settings_endpoint.py

```diff
@@ -55,31 +55,12 @@ async def get_settings(self, request):
     )
     @json_schema(schema(UpdateTriblerSettingsRequest={}))
     async def update_settings(self, request):
-        settings_dict = await request.json()
-        await self.parse_settings_dict(settings_dict)
-        self.tribler_config.write()
-        return RESTResponse({"modified": True})
-
-    async def parse_setting(self, section, option, value):
-        """
-        Set a specific Tribler setting. Throw a ValueError if this setting is not available.
-        """
-        # if section in self.config.config and option in self.config.config[section]:
-        self.tribler_config.__getattribute__(section).__setattr__(option, value)
-        # else:
-        #     raise ValueError(f"Section {section} with option {option} does not exist")
+        settings = await request.json()
+        self._logger.info(f'Received settings: {settings}')
+        self.config.update_from_dict(settings)
+        self.config.write()
 
-        # Perform some actions when specific keys are set
-        if section == "libtorrent" and (option == "max_download_rate" or option == "max_upload_rate"):
-            if self.download_manager:
-                self.download_manager.update_max_rates_from_config()
+        if self.download_manager:
```

Comment:
The condition `if section == "libtorrent" and (option == "max_download_rate" or option == "max_upload_rate"):` was always evaluating to True, making the check unnecessary.